### DISCUSSION TITLE
Resize with auto orient and ignore aspect ratio

### DISCUSSION THREAD

**Discussion by jungle-boogie** [ImageMagick/ImageMagick#7139] [Help] [answered]
### ImageMagick version

ImageMagick 7.1.1-28

### Operating system, version and so on

Windows 10 

### Description

This is similar to https://github.com/ImageMagick/ImageMagick/discussions/5772 and I made a comment there, but I don't know if it will get seen, since it's already been answered.

This works for landscape images:
`convert input.jpg -auto-orient -resize 1260x900!  output.jpg`

This work for portrait images:
`convert input.jpg -auto-orient -resize 900x1260!  output.jpg`

How can i have one command resize to either 5x7 or 7x5 depending on the orientation?
The auto-orient seems to work if I don't want to ignore the aspect ratio, but because I want to print the images to 5x7, I want to force the 1260x900.

This seems to return a 1 or 0 based on the w/h, how can I make this into a windows batch script to apply the correct conversion?
`convert input.jpg -format "%[fx:(w/h>1)?1:0]" info:`

Or is there a better way to resize images based on landscape/portrait mode?

Sample photos (deliberately low quality)

![P1010800_lq](https://github.com/ImageMagick/ImageMagick/assets/1111743/9cf234b9-b783-4689-861d-b006bc6af3e1)
![P1010807_lq](https://github.com/ImageMagick/ImageMagick/assets/1111743/a436dc3f-cf9a-4b9f-aab7-a5f40067094d)

*How would I resize to 5x7 or 7x5 respectively?*

**Comment by snibgo** [comment]
You can resize before auto-orient.

Does that do what you want?

**Reply by jungle-boogie** [reply]
How do you mean?

If I have a photo that was taken in portrait mode, what are the steps?

```
convert portait.jpg -auto-orient output.jpg
convert output.jpg resize ???
```

**Comment by snibgo** [comment]
As you have v7, I suggest using `magick` instead of `convert`.

You say you want `-resize 1260x900!` for landscapes, but `-resize 900x1200!` for portraits. Is that a typo, 1200 when you meant 1260? I assume so.

Non-square cameras make photos that are wider than they are tall. `-auto-orient` will rotate the image if the camera was held in portrait orientation.

So, before doing `-auto-orient`, the horizontal dimension is the largest, so `-resize 1260x900!` will do what you want. Then you can `-auto-orient`.
```
magick input.jpg -resize 1260x900! -auto-orient output.jpg
```

**Reply by jungle-boogie** [reply]
>  I assume so.

yes, typo and corrected. Thanks for pointing that out.

I think maybe I'm explaining my problem wrong - my fault.

if you take the second image above and use `magick input.jpg -resize 1260x900! -auto-orient output.jpg` this is the result:

![output](https://github.com/ImageMagick/ImageMagick/assets/1111743/d680f720-8f03-4fc7-9521-bc86874035f8)


The dimensions are correct, but it's obviously squished. It's even more obvious with a person:

![person2](https://github.com/ImageMagick/ImageMagick/assets/1111743/0c31c7d2-d359-43ae-ac80-38b2b38a636b)


Omitting `-auto-orient` doesn't seem to solve anything, either. 

I certainly can have two different directories of images - one for portrait and one for landscape, but I thought there would be something `magick` could do to see the orientation and scale it to the respective resolution.

**Comment by fmw42** [comment, marked as answer]
My command above works for me.  Try

```
magick image1.jpg -auto-orient -set option:wd "%[fx:(w>=h)?1260:900]!" -set option:ht "%[fx:(w>=h)?900:1260]!" -resize "%[wd]x%[ht]" output1.jpg

magick image2.jpg -auto-orient -set option:wd "%[fx:(w>=h)?1260:900]!" -set option:ht "%[fx:(w>=h)?900:1260]!" -resize "%[wd]x%[ht]" output2.jpg

```

**Reply by jungle-boogie** [reply]
That's great! thank you very much.

Can you explain what it is doing, especially the set options?

**Reply by fmw42** [reply]
Imagemagick 7 allows in-line computations so that you do not need to create variable ahead of time.  The -set option:XXX creates a new variable.  In this case, I create one for the resize width and one for the resize height. Each is a test whether w>=h in the input image. If it passes, I set the width or height for the output to one value and if it fails to another. The true case follows the ? and false case follows the \:
